stdlib/public/SDK/Foundation/NSRange.swift

```diff
@@ -105,18 +105,59 @@ extension NSRange {
 //===----------------------------------------------------------------------===//
 
 extension NSRange {
-    public init(_ x: Range<Int>) {
-        location = x.lowerBound
-        length = x.count
-    }
+  public init<R: RangeExpression>(_ rangeExpression: R)
+  where R.Bound: FixedWidthInteger, R.Bound.Stride : SignedInteger {
+    let range = rangeExpression.relative(to: 0..<R.Bound.max)
+    let start: Int = numericCast(range.lowerBound)
+    let end: Int = numericCast(range.upperBound)
+    self = NSRange(location: start, length: end - start)
+  }
+  
+  public init<R: RangeExpression, S: StringProtocol>(_ rangeExpression: R, in string: S)
+  where R.Bound == String.Index, S.Index == String.Index {
+    let range = rangeExpression.relative(to: string)
+    let start = range.lowerBound.samePosition(in: string.utf16)
+    let end = range.upperBound.samePosition(in: string.utf16)
+    let location = string.utf16.distance(from: string.utf16.startIndex, to: start)
+    let length = string.utf16.distance(from: start, to: end)
+    self = NSRange(location: location, length: length)
+  }
 
-    // FIXME(ABI)#75 (Conditional Conformance): this API should be an extension on Range.
-    // Can't express it now because the compiler does not support conditional
-    // extensions with type equality constraints.
-    public func toRange() -> Range<Int>? {
-        if location == NSNotFound { return nil }
-        return location..<(location+length)
-    }
+  @available(swift, deprecated: 4, renamed: "Range.init(_:)")
+  public func toRange() -> Range<Int>? {
+      if location == NSNotFound { return nil }
+      return location..<(location+length)
+  }
+}
+
+extension Range where Bound: BinaryInteger {
```

Comment:
Hmm, probably not. We don't generally encourage people to create the other range types without using the operators. `Range` is the currency type.